From 44ddfb20b3f8daa59c8e57058b8472bfdf9878df Mon Sep 17 00:00:00 2001
From: Varun Sharma <vsharma@habana.ai>
Date: Sun, 30 Jan 2022 21:00:29 +0200
Subject: [PATCH] [SW-72556] greco: pass btl version to host

This patch make changes to pass btl version along with
preboot version via bootinfo to host. This is required as
in greco we are currently using mixed flow and fw-loader
also checks for btl version when updating fw.

Change-Id: I8ef7483b60386b6db4850e96aa575dcdcc2cccdc
Signed-off-by: Varun Sharma <vsharma@habana.ai>
---

diff --git a/arch/greco/arc/common.c b/arch/greco/arc/common.c
index b3784b1..ae373f8 100644
--- a/arch/greco/arc/common.c
+++ b/arch/greco/arc/common.c
@@ -165,8 +165,10 @@
 	 * Due to DCCM ECC issue in arcboot rom, hybrid mode is enabled
 	 * for greco where execution starts from BTL and BTL starts
 	 * ROM flow on ARC1. Due to this, BTL_EN bit is set but still
-	 * ROM flow is executed. So aviod reading bootstrap register
-	 * here to and returning false for ARC flow.
+	 * ROM flow is executed. Since its mixed flow, we
+	 * will pass btl version to host
 	 */
-	return false;
+	u32 boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L);
+
+	return boot_strap & PSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L_BTL_EN_MASK;
 }
